web/pgadmin/static/js/helpers/ModalProvider.jsx

```diff
@@ -96,17 +96,43 @@ function alert(title, text, onOkClick, okLabel = gettext('OK')) {
 function confirm(title, text, onOkClick, onCancelClick, okLabel = gettext('Yes'), cancelLabel = gettext('No'), extras = null) {
   // bind the modal provider before calling
   this.showModal(title, (closeModal) => {
-    const onCancelClickClose = () => {
-      onCancelClick?.();
+    // Generalized function to handle button click and close the modal.
+    const handleClick = (callback) => {
+      callback?.();
       closeModal();
     };
-    const onOkClickClose = () => {
-      onOkClick?.();
+
+    return (
+      <AlertContent text={text} confirm onOkClick={() => handleClick(onOkClick)} onCancelClick={() => handleClick(onCancelClick)} okLabel={okLabel} cancelLabel={cancelLabel} extraButtons={extras?.(closeModal)} />
+    );
+  });
+}
+
+function confirmDelete(...args){
+  let [title, text, onOkClick, onCancelClick, okLabel = gettext('Yes'), cancelLabel = gettext('No')] = args;
+  // bind the modal provider before calling
+  this.showModal(title, (closeModal) => {
+    // Generalized function to handle button click and close the modal.
+    const handleClick = (callback) => {
+      callback?.();
       closeModal();
     };
-    const extraButtons =  extras?.(closeModal);
+
+    // Define additional buttons to be displayed in the modal.
+    const extraButtons = [
+      {
+        type: 'default',
+        icon: <CheckRoundedIcon />,
+        label: okLabel,
+        onclick: () => {
+          handleClick(onOkClick);
```

Comment:
same as above, why pass onOkClick when you can call directly?

Reply:
As handleClick is use to bind closeModal to both onOkClick and onCancelClick,
That's why I have passed onOkClick and onCancelClick.

Reply:
But handleClick is defined in the same function where onOkClick is directly available. Call it directly instead of passing and making it more complex.